src/yang/ietf-access-control-list.yang

```diff
@@ -444,25 +444,29 @@ module ietf-access-control-list {
               container tcp {
                 if-feature match-on-tcp;
                 uses pf:acl-tcp-header-fields;
-                choice source-port {
-                  container source-port-range-or-operator {
-                    uses pf:port-range-or-operator;
+                container source-port {
+                  choice source-port {
+                    case source-port-range-or-operator {
+                      uses pf:port-range-or-operator;
+                      description
+                        "Source port definition.";
+                    }
```

Comment:
I chose this method (as you found out yourself) so that I can avoid an extra case statement, if I put the choice statement first and put the container inside it. With your diffs, you are having to add an extra case statement, bloating the model.

Reply:
The point is that the data tree looks like crap with a port-range-or-operator container in it. 

My changes doesn't bloat the model. Once you add in the object reference stuff you are bound to repeat that container that you think saved you some space.